seclea_ai/seclea_ai.py

```diff
@@ -232,17 +238,40 @@ def upload_dataset(
                     )
             # upload all the datasets and transformations.
             for up_kwargs in upload_queue:
-                self._upload_dataset(**up_kwargs)  # TODO change
+                # self._upload_dataset(**up_kwargs)  # TODO change
+                self._storage_q.put(
+                    {
+                        "function": "upload_dataset",
+                        "project": self._project,
+                        "dataset": up_kwargs["dataset"],
+                        "dataset_name": up_kwargs["dataset_name"],
+                        "metadata": up_kwargs["metadata"],
+                        "parent_hash": up_kwargs["parent_hash"],
+                        "transformation": up_kwargs["transformation"],
+                    }
+                )
+                _thread = threading.Thread(target=self._file_processor.writer(self._storage_q))
+                _thread.start()
+                _sending_thread = threading.Thread(target=self._file_processor.sender())
+                _sending_thread.start()
```

Comment:
These should be started on init - there should be a corresponding _thread.join() in an end() method of seclea_ai.

Reply:
If we put thread.start in init function then it will try to execute queue when we create object of file_processor which has no data so execution will stop. We can't join it because thread already stopped.

Reply:
Okay that makes me think that the loop in the thread need to be fixed to not exit without data

Reply:
I have refactored, resolving this.